package NYT.NChunkClient.NProto;

import "yt_proto/yt/core/misc/proto/guid.proto";

////////////////////////////////////////////////////////////////////////////////

// Chunk Manager stuff.

message TChunkAddInfo
{
    required NYT.NProto.TGuid chunk_id = 1;

    // COMPAT(kvk1920): Remove after 23.2.
    optional NYT.NProto.TGuid location_uuid = 2;
    optional int32 location_directory_index = 7;
    optional uint32 location_index = 8;

    // COMPAT(kvk1920): Remove after real locations will be enabled everywhere.
    required int32 medium_index = 5;
    optional bool active = 3 [default = false];
    optional bool sealed = 4 [default = false];

    // COMPAT(kvk1920): Make `required` after real locations will be enabled everywhere.
    // NB: If real chunk locations are used this replica must be ignored.
    optional bool caused_by_medium_change = 6 [default = false];
}

message TChunkRemoveInfo
{
    required NYT.NProto.TGuid chunk_id = 1;

    // COMPAT(kvk1920): Remove after 23.2.
    optional NYT.NProto.TGuid location_uuid = 2;
    optional int32 location_directory_index = 5;
    optional uint32 location_index = 6;

    // COMPAT(kvk1920): Remove after real locations will be enabled everywhere.
    required int32 medium_index = 3;
    // COMPAT(kvk1920): Make `required` after real locations will be enabled everywhere.
    // NB: If real chunk locations are used this message must be ignored.
    optional bool caused_by_medium_change = 4 [default = false];
}

message TChunkReplicaAnnouncementRequest
{
    required NYT.NProto.TGuid chunk_id = 1;
    repeated fixed64 replicas = 2;

    //! If |true|, node should send announcement request only if sending lazy announcements
    //! is explicitly enabled.
    optional bool lazy = 3;

    //! If set, node should send announcement request after |delay|.
    optional int64 delay = 4; // TDuration

    //! If set, node should confirm receival upon next heartbeat.
    //! Such requests only come in incremental heartbeats.
    optional bool confirmation_needed = 5;
}

////////////////////////////////////////////////////////////////////////////////
